xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -607,8 +607,36 @@ void CLinuxRendererGLES::RenderUpdateVideo(bool clear, DWORD flags, DWORD alpha)
     CDVDVideoCodecIMXBuffer *buffer = m_buffers[m_iYV12RenderBuffer].IMXBuffer;
     if (buffer != NULL && buffer->IsValid())
     {
+      // this hack is needed to get the 2D mode of a 3D movie going
+      if (g_graphicsContext.GetStereoMode())
+        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_LEFT);
+
       ManageDisplay();
-      g_IMXContext.SetBlitRects(m_sourceRect, m_destRect);
+
+      if (g_graphicsContext.GetStereoMode())
+        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_OFF);
+
+      CRect dstRect(m_destRect);
+      CRect srcRect(m_sourceRect);
+      RENDER_STEREO_MODE stereo_mode = g_graphicsContext.GetStereoMode();
```

Comment:
Sorry, one thing: I'd move "stereo_mode ="  up and use it rather than call "GetStereoMode" 3 times